### ISSUE TITLE
How to enforce endless methods?

### ISSUE THREAD

**Issue by metalelf0** [rubocop/rubocop#11024]
Hi, I'm trying to get Rubocop to replace one-liners with endless methods in ruby 3.0.2.

I've tried different combinations of the `Style/EndlessMethod` and `Style/SingleLineMethods` cops, without any success.

--------

## Expected behavior

Suppose I have a file called `example.rb` with the following code:

```ruby
class Example
  def a_method
    "42"
  end
end
```

And the following `.rubocop.yml` file:

```yml
Style/EndlessMethod:
  Enabled: true
  EnforcedStyle: allow_single_line
```

I'd expect `rubocop -a example.rb` to report the method as endless-able, and correct it to:

```ruby
class Example
  def a_method = "42"
end
```

## Actual behavior

I only see other cops being ran, but there's no trace of the `Style/EndlessMethod` cop:

```
❯ rubocop -c .rubocop.yml -A example.rb --debug
configuration from /tmp/.rubocop.yml
Default configuration from /Users/andreaschiavini/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.36.0/config/default.yml
Inspecting 1 file
Scanning /tmp/example.rb
C

Offenses:

example.rb:3:1: C: Style/Documentation: Missing top-level documentation comment for class Example.
class Example
^^^^^^^^^^^^^
example.rb:5:5: C: [Corrected] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    "42"
    ^^^^

1 file inspected, 2 offenses detected, 1 offense corrected
Finished in 0.13531199999852106 seconds
```

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

- create the ruby file above
- create the provided `.rubocop.yml` enabling the endless method cop
- run it

## RuboCop version

```
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.0.2) [arm64-darwin21]
```

**Comment by jonas054** [comment]
There's no support currently for the functionality you want, but it should be possible to add. I also happen to think it makes sense to be able to enforce endless method definitions. Because I like them.

Let's say we want to support autocorrecting from
```ruby
class Example
  def a_method
    "42"
  end
end
```
via
```ruby
class Example
  def a_method; "42" end
end
```
to
```ruby
class Example
  def a_method = "42"
end
```
It would require changes in both `Style/SingleLineMethods` and `Style/EndlessMethod`. The easier of these two is `Style/EndlessMethod` becuase it has an `EnforcedStyle` configuration parameter and we can just add a supported style called `require_single_line` (meaning: require endless method definitions for methods that are defined on a single line). This would be backwards compatible. 

Fixing `Style/SingleLineMethods` is trickier bacause we don't have an `EnforcedStyle` there, only an `AllowIfMethodIsEmpty`. So how do we set up new config parameters to be able to enforce single line methods in a backwards compatible way? There's also a cop called `Layout/RedundantLineBreak`, which is disabled by default, but seems to be a candidate for enforcing single line methods too, although it doesn't do this currently.

**Comment by vlad-pisanov** [comment]
👍 +1 would love to see this enhancement as well

**Comment by Abdullah-l** [comment]
I would to see endless methods enforced by rubocop
